src/marks/auto.js

```diff
@@ -142,43 +142,41 @@ function autoImpl(data, options) {
     z: Z,
     r: S ?? undefined // treat null size as undefined for default constant radius
   };
-  let transform;
+  let transformImpl;
   let transformOptions = {[colorMode]: colorReduce ?? undefined, r: sizeReduce ?? undefined};
   if (xReduce != null && yReduce != null) {
     throw new Error(`cannot reduce both x and y`); // for now at least
   } else if (yReduce != null) {
     transformOptions.y = yReduce;
-    transform = isOrdinal(X) ? groupX : binX;
+    transformImpl = isOrdinal(X) ? groupX : binX;
   } else if (xReduce != null) {
     transformOptions.x = xReduce;
-    transform = isOrdinal(Y) ? groupY : binY;
+    transformImpl = isOrdinal(Y) ? groupY : binY;
   } else if (colorReduce != null || sizeReduce != null) {
     if (X && Y) {
-      transform = isOrdinal(X) && isOrdinal(Y) ? group : isOrdinal(X) ? binY : isOrdinal(Y) ? binX : bin;
+      transformImpl = isOrdinal(X) && isOrdinal(Y) ? group : isOrdinal(X) ? binY : isOrdinal(Y) ? binX : bin;
     } else if (X) {
-      transform = isOrdinal(X) ? groupX : binX;
+      transformImpl = isOrdinal(X) ? groupX : binX;
     } else if (Y) {
-      transform = isOrdinal(Y) ? groupY : binY;
+      transformImpl = isOrdinal(Y) ? groupY : binY;
     }
   }
-  if (transform) {
-    if (transform === bin || transform === binX) markOptions.x = {value: X, ...xOptions};
-    if (transform === bin || transform === binY) markOptions.y = {value: Y, ...yOptions};
-  }
+
+  // When using the bin transform, pass through additional options (e.g., thresholds).
+  if (transformImpl === bin || transformImpl === binX) markOptions.x = {value: X, ...xOptions};
+  if (transformImpl === bin || transformImpl === binY) markOptions.y = {value: Y, ...yOptions};
 
   // If zero-ness is not specified, default based on whether the resolved mark
   // type will include a zero baseline.
   if (xZero === undefined)
     xZero =
       X &&
-      transform !== bin &&
-      transform !== binX &&
+      !(transformImpl === bin || transformImpl === binX) &&
```

Comment:
Is the point of this change that they're mutually exclusive and `!(a || b)` can short-circuit faster than `!a && !b` or something?